packages/components/src/components/assets/AssetShareIndicator/AssetShareIndicator.tsx

```diff
@@ -101,15 +195,31 @@ export const AssetShareIndicator = ({
     size = 32,
     percentageShare,
     index,
+    coingeckoId = '',
+    contractAddress,
+    quality = 'medium',
     ...rest
-}: AssetShareIndicatorProps) => (
-    <Container className={className}>
-        <CoinLogo symbol={symbol} size={size} {...rest} />
-        <ProgressCircle
-            symbol={symbol}
-            size={size}
-            percentageShare={percentageShare}
-            index={index}
-        />
-    </Container>
-);
+}: AssetShareIndicatorProps) => {
+    const iconUrl = useAssetUrl(coingeckoId, contractAddress);
+    const coinLogoSize = !symbol ? QUALITY_SIZE[quality] : size;
+
+    return (
+        <Container className={className} $size={coinLogoSize * 2}>
```

Comment:
I undestand that it should be bigger but how exactly do we know that it should be doubled?

Reply:
Ideally it should work without setting up exact size but it might lead to some visual issues

Reply:
> I understand that it should be bigger but how exactly do we know that it should be doubled?

I had to figure out that the size of the whole component is icon size * 2. Additionally, the original implementation was buggy - the width and height were hardcoded in the Container component, and that lead to too big empty space in case of using smaller icon (smaller than 24px) - which we need now. Now with the actual approach, we make sure the size will always be correct and corresponding with the size of the icon we decide to use.

Reply:
However... I realized we might break the actual usage of AssetShareIndicator in the code if using this change, occurrences of the component that use default 32px icon size and 48px size of the container - which naturally created some nice space around. Ideally, we should keep this while meet the new expectations. Not sure how I'm gonna solve it.